File: sdk/python/config/AvatarMessage.py
############################################################################
#
#  Copyright 2023 Paphus Solutions Inc.
#
#  Licensed under the Eclipse Public License, Version 1.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#
#      http://www.eclipse.org/legal/epl-v10.html
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.
#
############################################################################
from config.Config import Config
from util.Utils import Utils, Writer
class AvatarMessage(Config):
    message: str
    avatar: str
    emote: str
    action: str
    pose: str
    speak: bool
    voice: str
    format: str
    hd: bool
    
    def __init__(self):
        super().__init__()
        self.message = None
        self.avatar = None
        self.emote = None
        self.action = None
        self.pose = None
        self.speak = False
        self.voice = None
        self.format = None
        self.hd = False
        
        
    def toXML(self):
        writer = Writer("<avatar-message")
        writer = self.writeCredentials(writer)
        if(self.avatar != None):
            writer.append(" avatar=\"" + self.avatar + "\"")
        
        if(self.emote != None):
            writer.append(" emote=\"" + self.emote + "\"")
            
        if(self.action != None):
            writer.append(" action=\"" + self.action + "\"")
        
        if(self.pose != None):
            writer.append(" pose=\"" + self.pose + "\"")
        
        if(self.format != None):
            writer.append(" format=\"" + self.format + "\"")
        
        if(self.voice != None):
            writer.append(" voice=\"" + self.voice + "\"")
        
        if(self.speak != False):
            writer.append(" speak=\"true\"")
            
        if(self.hd != False):
            writer.append(" hd=\"true\"")
            
        writer.append(">")
        
        if(self.message): 
            writer.append("<message>")
            writer.append(Utils.escapeHTML(self.message))
            writer.append("</message>")
            
        writer.append("</avatar-message>")
        return writer

File: sdk/nodejs/src/config/AvatarMessage.ts
/*
*  Copyright 2023 Paphus Solutions Inc.
*
*  Licensed under the Eclipse Public License, Version 1.0 (the "License");
*  you may not use this file except in compliance with the License.
*  You may obtain a copy of the License at
*
*      http://www.eclipse.org/legal/epl-v10.html
*
*  Unless required by applicable law or agreed to in writing, software
*  distributed under the License is distributed on an "AS IS" BASIS,
*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
*  See the License for the specific language governing permissions and
*  limitations under the License.
*/

import { XMLWriter } from "../util/Utils";
import Config from "./Config";

export default class AvatarMessage extends Config {
    message?: string
    avatar?:string
    emote?:string
    action?:string
    pose?:string
    speak?:boolean
    voice?:string
    format?: string
    hd?:boolean
    constructor(){
        super()
    }
    toXML(): string{
        let writer = new XMLWriter("<avatar-message")
        this.writeCredentials(writer)
        writer.appendAttribute('avatar', this.avatar)
        writer.appendAttribute('emote', this.emote)
        writer.appendAttribute('action', this.action)
        writer.appendAttribute('pose', this.pose)
        writer.appendAttribute('format', this.format)
        writer.appendAttribute('voice', this.voice)
        writer.appendAttribute('speak', this.speak)
        writer.appendAttribute('hd', this.hd)
        writer.append(">")
        writer.appendElement('message', this.message, true)
        writer.append('</avatar-message>')
        return writer.toString()
    }
}